stats_overlay.js

```diff
@@ -0,0 +1,394 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * @fileoverview A statistics visualiser.
+ */
+
+ goog.provide('shaka.StatsOverlay');
+
+/** @constructor
+ * @export
+*/
+shaka.StatsOverlay = function() {
+
+  /** @private {?shaka.player.Player} */
+  this.player_;
+
+  /** @private {?HTMLElement} */
+  this.canvas_;
+
+  /** @private {?HTMLVideoElement} */
+  this.video_;
+
+  /** @private {?number} */
+  this.width_;
+
+  /** @private {?number} */
+  this.height_;
+
+  /** @private {number} */
+  this.maxY_ = 1;
+
+  /** @private {number} */
+  this.maxBufferLength_ = 1;
+
+  /** @private {?number} */
+  this.startTimestamp_;
+
+  /** @private {?number} */
+  this.currentXOffset_;
+
+  /** @private {?number} */
+  this.refreshIntervalId_;
+
+  /** @private {?Array} */
+  this.bufferLengthHistory_;
+
+};
+
+
+/**
+ * Initializes the stats ovelay.
+ * @param {!shaka.player.Player} player
+ * @param {?HTMLElement} canvas
+*  @param {!HTMLVideoElement} video The video element.
+ * @export
+ */
+shaka.StatsOverlay.prototype.init = function(player, canvas, video) {
+
+  this.player_ = player;
+  this.canvas_ = canvas;
+  this.video_ = video;
+
+  this.width_ = canvas.width;
+  this.height_ = canvas.height - 5;
+
+  this.reset();
+};
+
+
+/**
+ * Converts a timestamp to an x coordinate
+ * @private
+ * @param {?number} timestamp
+ * @return {number}
+ */
+shaka.StatsOverlay.prototype.convertTimestampToXCoordinate_ =
+    function(timestamp) {
+  return Math.round(timestamp - this.startTimestamp_ + this.currentXOffset_) +
+         0.5;
+};
+
+
+/**
+ * Converts bits to the Y Coordinate
+ * Updated the running maxY_ to calculate the correct scale
+ * @private
+ * @param {?number} bits
+ * @return {number}
+ */
+shaka.StatsOverlay.prototype.convertBitsToYCoordinate_ = function(bits) {
+
+  // update the running scale
+  if (bits && bits > this.maxY_) {
+    this.maxY_ = bits;
+
+    // update the scaleY
+    this.scaleY_ = - (this.height_ / (this.maxY_ * 1.1));
+  }
+
+  // fudge the coordindate to ensure that the paths produce clean lines
+  return Math.round((bits * this.scaleY_) + this.height_) + 0.5;
+};
+
+/**
+ * Converts bits to the Y Coordinate
+ * Updated the running maxY_ to calculate the correct scale
+ * @private
+ * @param {?number} bufferLength
+ * @return {number}
+ */
+shaka.StatsOverlay.prototype.convertBufferLengthToYCoordinate_ = function(bufferLength) {
+
+  // update the running scale
+  if (bufferLength && bufferLength > this.maxBufferLength_) {
+    this.maxBufferLength_ = bufferLength;
+
+    // update the bufferLength scaleY
+    this.scaleBufferLengthY_ = - (this.height_ / (this.maxBufferLength_ * 1.1));
+  }
+
+  // fudge the coordindate to ensure that the paths produce clean lines
+  return Math.round((bufferLength * this.scaleBufferLengthY_) + this.height_) + 0.5;
+};
+
```

Comment:
It seems there are several style problems in this code.  Please run ./build/lint.sh on this and clean up the issues it reports.